client/src/client.rs

```diff
@@ -248,16 +250,41 @@ pub struct ResultSet<T> {
     client_cursor: usize,
 }
 
-impl<T: Clone> Iterator for ResultSet<T>
+pub struct SeekResultSet<'a, T, R>
```

Comment:
Is there a particular reason this struct is called `Seek`? Also, why does it always uses `SyncRequestManager`? Is it more like `SyncResultSet`?